org.bardsoftware.eclipsito/src/org/bardsoftware/impl/eclipsito/BootImpl.java

```diff
@@ -42,24 +42,21 @@ public void run(String application, List<File> modulesDirs, String descriptorPat
 
   public File getVersionDir(List<File> modulesDir) {
     assert modulesDir != null : "No modules directories";
-    String version = null;
-    String pluginsPath = null;
     if (modulesDir.isEmpty()) {
       return null;
     }
+    String version = null;
+    File versionDir = null;
     for (File dir : modulesDir) {
       String dirVersion = getVersionNumber(dir);
       if (version == null
               || (dirVersion != null && version.compareTo(dirVersion) < 0)) {
         version = dirVersion;
-        pluginsPath = dir.getPath() + File.separator + dirVersion;
+        versionDir = new File(dir, dirVersion);
+        assert versionDir.exists() : String.format("Directory %s doesn't exist", versionDir.getAbsolutePath());
+        assert versionDir.isDirectory() && versionDir.canRead() : String.format("File %s is not a directory or is not readable", versionDir.getAbsolutePath());
```

Comment:
I mean, when it is assert then it just fails. We probably want to be a little bit more robust and log this issue, but do not fail. E.g. imagine that for whatever reason user deleted his ~/.ganttproject.d/plugins/3.0.1 folder but left VERSION file. He has plugins/3.0.0 folder in the GanttProject install directory though. With this assert GanttProject will fail to start and user will be crying WTF because he is unable to do his job. We'll help him of course but it will take time